Source/levels/drlg_l1.cpp

```diff
@@ -1003,44 +986,31 @@ void Substitution()
 
 void FillChambers()
 {
+	Point chamber1 { 14, 0 };
+	Point chamber3 { 14, 28 };
+	Point hall1 { 18, 12 };
+	Point hall2 { 18, 26 };
 	if (!VerticalLayout) {
-		if (HasChamber1)
-			GenerateChamber({ 0, 14 }, false, false, false, true);
+		std::swap(chamber1.x, chamber1.y);
+		std::swap(chamber3.x, chamber3.y);
+		std::swap(hall1.x, hall1.y);
+		std::swap(hall2.x, hall2.y);
+	}
 
-		if (!HasChamber3)
-			GenerateChamber({ 14, 14 }, false, false, true, false);
-		else if (!HasChamber1)
-			GenerateChamber({ 14, 14 }, false, false, false, true);
-		else if (HasChamber1 && HasChamber2 && HasChamber3)
-			GenerateChamber({ 14, 14 }, false, false, true, true);

```

Comment:
The bug was here, i was always connecting chamber 2 to at least one other chamber. But instead of the multiple if's we can simply use the variable as the paramter which makes things a lot cleaner. At that point it also became feasible to merge vertical/horizontal layout and just pass on the layout direction instead of alternating between parameters 1&2 and 3&4 for the chambers.